diff --git a/chrome/browser/search_engines/template_url_service_factory.cc b/chrome/browser/search_engines/template_url_service_factory.cc
index 542e397bcb868995e921ecc1bd8a518a9272d398..472ad4cf58fb6ff67d6afad4bf46c6f18302ad0c 100644
--- a/chrome/browser/search_engines/template_url_service_factory.cc
+++ b/chrome/browser/search_engines/template_url_service_factory.cc
@@ -82,7 +82,12 @@ void TemplateURLServiceFactory::RegisterProfilePrefs(
 
 content::BrowserContext* TemplateURLServiceFactory::GetBrowserContextToUse(
     content::BrowserContext* context) const {
+#if defined(BRAVE_CHROMIUM_BUILD)
+  // To make different service for normal and incognito profile.
+  return chrome::GetBrowserContextOwnInstanceInIncognito(context);
+#else
   return chrome::GetBrowserContextRedirectedInIncognito(context);
+#endif
 }
 
 bool TemplateURLServiceFactory::ServiceIsNULLWhileTesting() const {
